scripts/setup-centos9.sh

```diff
@@ -99,9 +106,13 @@ function install_lzo {
 function install_boost {
   wget_and_untar https://github.com/boostorg/boost/releases/download/${BOOST_VERSION}/${BOOST_VERSION}.tar.gz boost
   (
-   cd boost
-   ./bootstrap.sh --prefix=/usr/local
-   ./b2 "-j$(nproc)" -d0 install threading=multi --without-python
+    # Note, this will use the guessed toolset. If both gcc and clang-15 are present it will pick gcc.
+    # There is no easy way to use clang++-15 if it doesn't exist as clang++.
+    # Therefore, in order to use clang-15, one needs to install it as an alternative, set it as the default, and then
+    # select the --with-toolset=clang for boostrap script and toolset=clang for the b2 script.
```

Comment:
Let's add `--with-toolset=clang` to bootstrap and `toolset=clang` for the b2 script when `USE_CLANG` is used.
The error then will force users to set up the clang++ correctly.

Reply:
So the problem is then that this will fail on  fresh install. This assumes a user has not had anything installed previously (fresh setup) and sets USE_CLANG. In that case clang-15 isn't symlinked to clang after installation (which is why I had set up the alternatives before). That is a problem. So the experience will be that this will fail.

Reply:
The default behavior will be to build boost using GCC correct? This is not ideal.
It's okay to error. If we document this in the README, users will understand how to overcome the error.

Reply:
> In that case clang-15 isn't symlinked to clang after installation 

Apologies, why is this the case ?

Reply:
That's likely because the default clang version on Centos9 was Clang17 (when I started this) and is now Clang18. So just installing "clang" will result in the system default version installation symlinked to clang and clang++.

Reply:
> The default behavior will be to build boost using GCC correct
Yes, because we also install GCC and so it finds it instead.

The README would have to document setting up the alternatives. At this point Clang15 is already installed. Otherwise, if a newer Clang (system default) is present it will use that version (and not Clang15) to perform the build.

Reply:
Doesnt boost check for the compiler specified via CXX env variables ? That is what they should be using and if thats not set, go about using latest compiler.

Reply:
No. Unfortunately, it doesn't. CC/CXX are CMake environment variables. And the boost boostrap script specifically tries to use clang/clang++ to check for the toolset.

This is what it looks like:
```
++ export CC=/usr/bin/clang-15
++ CC=/usr/bin/clang-15
++ export CXX=/usr/bin/clang++-15
++ CXX=/usr/bin/clang++-15
++ [[ 1 -ne 0 ]]
++ for cmd in "$@"
++ run_and_time install_boost
++ install_boost
++ wget_and_untar https://github.com/boostorg/boost/releases/download/boost-1.84.0/boost-1.84.0.tar.gz boost
++ local URL=https://github.com/boostorg/boost/releases/download/boost-1.84.0/boost-1.84.0.tar.gz
++ local DIR=boost
++ mkdir -p boost
++ pushd boost
~/deps/boost ~/deps
++ curl -L https://github.com/boostorg/boost/releases/download/boost-1.84.0/boost-1.84.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  118M  100  118M    0     0  82.8M      0  0:00:01  0:00:01 --:--:--  124M
++ tar -xz --strip-components=1 -f boost.tar.gz
++ popd
~/deps
++ cd boost
++ ./bootstrap.sh --prefix=/usr/local --with-toolset=clang
Building B2 engine..

A C++11 capable compiler is required for building the B2 engine.
Toolset 'clang' does not appear to support C++11.

> clang++ -x c++ -std=c++11  check_cxx11.cpp
./tools/build/src/engine/build.sh: 120: clang++: not found
```

Reply:
I found a solution to the problem.

Reply:
Finally, found a solution to the problem.

Reply:
Thank you @czentgr , is the solution what you described above ?

Reply:
No but it was highlighting some customization. Only problem was that bootstrap output and b2 jam files didn't quite work in the desired way. I needed it to use clang 15 in the bootstrap - which I did. But the generated jam file required a separate definition for the "new" clang-15" toolset when I wanted was to use the clang compiler of version 15 because it has code to automatically build the right command.
And providing a custom user defined jam file with the compiler (from the b2 doc) also didn't help because it was using the generated version which loaded the compiler differently. So I ended up changing the generated jam file to actually load the known compiler in the specific version we wanted and that works. Code is this
```
      ./bootstrap.sh --prefix=/usr/local --with-toolset="clang-15"
      # Switch the compiler from the clang-15 toolset which doesn't exist (clang-15.jam) to
      # clang of version 15 when toolset clang-15 is used.
      # This reconciles the project-config.jam generation with what the b2 build system allows for customization.
      sed -i 's/using clang-15/using clang : 15/g' project-config.jam
      ${SUDO} ./b2 "-j$(nproc)" -d0 install threading=multi toolset=clang-15 --without-python
 ```